File: filesystem/src/main/java/com/nytimes/android/external/fs3/SourceFileWriter.java
package com.nytimes.android.external.fs3;

import com.nytimes.android.external.fs3.filesystem.FileSystem;
import com.nytimes.android.external.store3.base.DiskWrite;
import com.nytimes.android.external.store3.base.impl.BarCode;

import okio.BufferedSource;

public class SourceFileWriter extends FSWriter<BarCode> implements DiskWrite<BufferedSource, BarCode> {

    public SourceFileWriter(FileSystem fileSystem) {
        super(fileSystem, new BarCodePathResolver());
    }

}

File: filesystem/src/main/java/com/nytimes/android/external/fs3/SourceFileWriter.kt
package com.nytimes.android.external.fs3

import com.nytimes.android.external.fs3.filesystem.FileSystem
import com.nytimes.android.external.store3.base.DiskWrite
import com.nytimes.android.external.store3.base.impl.BarCode

import okio.BufferedSource

class SourceFileWriter @JvmOverloads
constructor(fileSystem: FileSystem, pathResolver:
PathResolver<BarCode> = BarCodePathResolver()) :
        FSWriter<BarCode>(fileSystem, pathResolver), DiskWrite<BufferedSource, BarCode>
